packages/components/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,225 @@
+   */
+  onMouseOver?: (evt: MouseEvent) => void;
+  /**
+   * It sets the "preferred" position of the Tooltip
+   */
+  place?: TooltipPlace;
+  /**
+   * Class names to be appended to the className prop of the Tooltip’s target
+   */
+  targetWrapperClassName?: string;
+  /**
+   * An ID used for testing purposes applied as a data attribute (data-test-id)
+   */
+  testId?: string;
+  /**
+   * Boolean to control whether or not to render the tooltip in a React Portal.
+   * Rendering content inside a Portal allows the tooltip to escape the bounds
+   * of its parent while still being positioned correctly. Using a Portal is
+   * necessary if an ancestor of the tooltip hides overflow.
+   *
+   * Defaults to `false`
+   */
+  usePortal?: boolean;
+}
+
+export const Tooltip = ({
+  children,
+  className,
+  as: HtmlTag = 'span',
+  content,
+  id,
+  isVisible = false,
+  hideDelay = 0,
+  onBlur,
+  onFocus,
+  onMouseLeave,
+  onMouseOver,
+  targetWrapperClassName,
+  maxWidth = 360,
+  testId = 'cf-ui-tooltip',
+  place = 'auto',
+  usePortal = false,
+  ...otherProps
+}: TooltipProps) => {
+  const [show, setShow] = useState(false);
+
+  const elementRef = useRef(null);
+  const popperRef = useRef(null);
+  const [arrowRef, setArrowRef] = useState<HTMLSpanElement | null>(null);
+  const { styles: popperStyles, attributes, forceUpdate } = usePopper(
+    elementRef.current,
+    popperRef.current,
+    {
+      placement: place,
+      modifiers: [
+        {
+          name: 'arrow',
+          options: {
+            element: arrowRef,
+            padding: parseFloat(tokens.borderRadiusSmall),
+          },
+        },
+        {
+          name: 'offset',
+          options: {
+            offset: [0, 10],
+          },
+        },
+      ],
+    },
+  );
+
+  // necessary to update tooltip position in case the content is being updated
+  useEffect(() => {
+    if (forceUpdate !== null) {
+      forceUpdate();
+    }
+  }, [content, forceUpdate]);
+
+  const [isHoveringTarget, setIsHoveringTarget] = useState(false);
+  const [isHoveringContent, setIsHoveringContent] = useState(false);
+  useEffect(() => {
+    setShow(isHoveringContent || isHoveringTarget);
+  }, [isHoveringTarget, isHoveringContent]);
+
+  useEffect(() => {
+    if (isVisible) setShow(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const contentMaxWidth =
+    typeof maxWidth === 'string' ? maxWidth : `${maxWidth}px`;
+
+  const contentStyles: CSSProperties = {
+    zIndex: tokens.zIndexTooltip,
+    maxWidth: contentMaxWidth,
+    ...popperStyles.popper,
+  };
+
+  if (!content) {
+    return <HtmlTag className={targetWrapperClassName}>{children}</HtmlTag>;
+  }
+
+  const tooltip = (
+    <span
+      id={id}
+      ref={popperRef}
+      role="tooltip"
+      style={contentStyles}
+      className={cn(styles.tooltip, className)}
+      data-test-id={testId}
```

Comment:
Using our `Primitive` component here would set some of these props for us.

Reply:
+1 on this one